src/targets/gpu/time_op.cpp

```diff
@@ -51,21 +50,19 @@ time_op(context& ictx, operation op, const std::vector<shape>& inputs, int n)
     auto output           = op.compute_shape(inputs);
     op.finalize(ctx, output, inputs);
     auto args = generate_arguments(inputs);
-    auto run  = [&] {
-        op.compute(ctx, output, args);
-        ctx.finish();
-    };
-    gctx.enable_perf_measurement();
+    auto start = context::create_event();
+    auto stop  = context::create_event();
+    auto run   = [&] { op.compute(ctx, output, args); };
     run();
-    double host_time   = 0.0;
-    double device_time = 0.0;
+    gctx.get_stream().record(start.get());
     for(auto i : range(n))
     {
         (void)i;
-        host_time += time<milliseconds>(run);
-        device_time += gctx.get_elapsed_ms();
+        run();
     }
-    return std::make_pair(host_time / n, device_time / n);
+    gctx.get_stream().record(stop.get());
+    gctx.finish();
+    return context::get_elapsed_ms(start.get(), stop.get()) / n;
```

Comment:
I am not sure why it wouldn't work, but either case we wouldn't be using mult-stream execution to time operators.

Reply:
it wouldn't work with multi-stream i think because `hipEventElapsedTime` records time between start and stop events. both those events are put on the default stream.  If any of the op was using some other streams then it wouldn't measure correctly

Reply:
The only op that should change the default stream would be `gpu::set_stream`, which it doesnt make sense for us to do timing for.